### PR TITLE
fix: ConfirmHouseResizeGump, ConfirmDryDockGump to StaticGump

### PR DESCRIPTION
**Pull request by IcculusC**
### Summary
- Converts ConfirmHouseResizeGump, ConfirmDryDockGump to StaticGump
- ConfirmHouseResizeGump now supports the account gold system.

### CODE REVIEW

File: Projects/UOContent/Gumps/ConfirmHouseResize.cs [hunk 1]
```diff
@@ -3,170 +3,171 @@
...
-            AddHtmlLocalized(40, 250, 170, 20, 1011036, 0x7FFF);  // OKAY
-            AddHtmlLocalized(240, 250, 170, 20, 1011012, 0x7FFF); // CANCEL
-        }
+        mobile.CloseGump<ConfirmHouseResize>();
+    }
+
+    protected override void BuildLayout(ref StaticGumpBuilder builder) {
+        builder.SetNoClose();
+
+        builder.AddPage(0);
```
Comment (IcculusC) [inline Projects/UOContent/Gumps/ConfirmHouseResize.cs:24]:
```suggestion
        builder.AddPage();
```

File: Projects/UOContent/Multis/Boats/ConfirmDryDockGump.cs [hunk 2]
```diff
@@ -1,40 +1,43 @@
...
-            m_From = from;
-            m_Boat = boat;
+        _from.CloseGump<ConfirmDryDockGump>();
 
-            m_From.CloseGump<ConfirmDryDockGump>();
+    }
 
-            AddPage(0);
+    protected override void BuildLayout(ref StaticGumpBuilder builder) {
+        builder.AddPage(0);
```
Comment (IcculusC) [inline Projects/UOContent/Multis/Boats/ConfirmDryDockGump.cs:21]:
```suggestion
        builder.AddPage();
```

File: Projects/UOContent/Gumps/ConfirmHouseResize.cs [hunk 3]
```diff
@@ -3,170 +3,171 @@
...
+            else
             {
-                m_Mobile.CloseGump<ConfirmHouseResize>();
-                m_Mobile.SendGump(new HouseGumpAOS(HouseGumpPageAOS.Customize, m_Mobile, m_House));
+                _mobile.SendLocalizedMessage(501320); // Only the house owner may do this.
             }
         }
+        else if (info.ButtonID == 0)
+        {
+            _mobile.CloseGump<ConfirmHouseResize>();
```
Comment (kamronbatman) [inline Projects/UOContent/Gumps/ConfirmHouseResize.cs:169]: I think this is redundant, cause at this point, the gump is closed (due to the response).

File: Projects/UOContent/Gumps/ConfirmHouseResize.cs [hunk 4]
```diff
@@ -3,170 +3,171 @@
...
+        builder.AddButton(10, 250, 0xFA5, 0xFA7, 1);
+        builder.AddButton(210, 250, 0xFA5, 0xFA7, 0);
+        builder.AddHtmlLocalized(40, 250, 170, 20, 1011036, 0x7FFF);  // OKAY
+        builder.AddHtmlLocalized(240, 250, 170, 20, 1011012, 0x7FFF); // CANCEL
+    }
 
-        public override void OnResponse(NetState state, in RelayInfo info)
+    public override void OnResponse(NetState state, in RelayInfo info)
+    {
+        if (info.ButtonID == 1 && !_house.Deleted)
```
Comment (kamronbatman) [inline Projects/UOContent/Gumps/ConfirmHouseResize.cs:53]: Let's simplify OnResponse with a fail fast.

```cs
if (info.ButtonID > 1 || _house.Deleted)
{
    return;
}
```

File: Projects/UOContent/Multis/Boats/ConfirmDryDockGump.cs [hunk 5]
```diff
@@ -1,40 +1,43 @@
...
 using Server.Network;
 
-namespace Server.Multis
+namespace Server.Multis;
+
+public class ConfirmDryDockGump : StaticGump<ConfirmDryDockGump>
 {
-    public class ConfirmDryDockGump : Gump
+    private readonly BaseBoat _boat;
+    private readonly Mobile _from;
```
Comment (kamronbatman) [inline Projects/UOContent/Multis/Boats/ConfirmDryDockGump.cs:9]: For these gumps, we don't even need to store the `_from`. Let's use `state.Mobile` (when it is not null).

File: Projects/UOContent/Gumps/ConfirmHouseResize.cs [hunk 6]
```diff
@@ -3,170 +3,171 @@
...
 using Server.Network;
 
-namespace Server.Gumps
+namespace Server.Gumps;
+
+public class ConfirmHouseResize : StaticGump<ConfirmHouseResize>
 {
-    public class ConfirmHouseResize : Gump
+    private readonly BaseHouse _house;
+    private readonly Mobile _mobile;
```
Comment (kamronbatman) [inline Projects/UOContent/Gumps/ConfirmHouseResize.cs:11]: Same as the other comment, we don't need to store this.

**Pull request merged**